pkg/data/generic.go

```diff
@@ -49,12 +54,30 @@ func New(kubeconfig *rest.Config, opa opa_client.Data, ns types.ResourceType) *G
 	return NewFromInterface(client, opa, ns)
 }
 
+type Option func(s *GenericSync)
+
 // NewFromInterface returns a new GenericSync that can be started.
-func NewFromInterface(client dynamic.Interface, opa opa_client.Data, ns types.ResourceType) *GenericSync {
-	return &GenericSync{
-		client: client,
-		ns:     ns,
-		opa:    opa.Prefix(ns.Resource),
+func NewFromInterface(client dynamic.Interface, opa opa_client.Data, ns types.ResourceType, opts ...Option) *GenericSync {
```

Comment:
It's a technicality, but this is a breaking change: Before, you could have done

```go
var f func(dynamic.Interface, opa_client.Data, types.ResourceType) *data.GenericSync = data.NewFromInterface
```
now, it would change because the function has gotten another argument.

The safer alternative would be to introduce a new entrypoint, ` NewFromInterfaceWithOpts`.

Reply:
yep, I noticed, but `NewFromInterface` was introduced short ago (in c5723577465cfa9988cffdf94eeeb50d482d5723) and has never been featured in a release AFAIK. So it would not be a breaking change for anyone using a tagged version of kube-mgmt, and I'm not sure if it's worth adding another function because of that.

Reply:
I'm afraid I've released 4.1.0 yesterday 😅 

But they said we might no expect many library users here so it's possibly ok a anyways. What do you think?

Reply:
Ouch! :D . I am preparing another PR with test cases for `pkg.configmap.Sync`, and it will add a similar `NewFromInterface` function to that package too. So the name we decide for it in this PR will spread to the `configmap` package, for consistency. 

Only because of that, I'd prefer to stick with the shorter name, considering it will rarely impact any user of kube-mgmt.

Reply:
Yeah it's ok. As I said, it's just a technicality at this point. 👍